tests/components/homekit/test_config_flow.py

```diff
@@ -0,0 +1,259 @@
+"""Test the HomeKit config flow."""
+from asynctest import patch
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.homekit.const import DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_NAME, CONF_PORT
+
+from tests.common import MockConfigEntry
+
+
+def _mock_config_entry_with_options_populated():
+    """Create a mock config entry with options populated."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_NAME: "mock_name", CONF_PORT: 12345},
+        options={
+            "filter": {
+                "include_domains": [
+                    "fan",
+                    "vacuum",
+                    "media_player",
+                    "climate",
+                    "alarm_control_panel",
+                ],
+                "exclude_entities": ["climate.front_gate"],
+            },
+            "auto_start": False,
+            "safe_mode": False,
+            "zeroconf_default_interface": True,
+        },
+    )
+
+
+async def test_user_form(hass):
+    """Test we can setup a new instance."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5453535)